misk-aws/src/main/kotlin/misk/dynamodb/FakeDynamoDbModule.kt

```diff
@@ -0,0 +1,73 @@
+package misk.dynamodb
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider
+import com.amazonaws.auth.BasicAWSCredentials
+import com.amazonaws.client.builder.AwsClientBuilder
+import com.amazonaws.regions.Regions
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper
+import com.amazonaws.services.dynamodbv2.document.DynamoDB
+import com.amazonaws.services.dynamodbv2.model.ProvisionedThroughput
+import com.amazonaws.services.dynamodbv2.model.ResourceInUseException
+import com.google.inject.Provides
+import misk.cloud.aws.FakeAwsEnvironmentModule
+import misk.inject.KAbstractModule
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+
+internal class FakeDynamoDbModule(
+  private val entities: List<Pair<String, KClass<*>>>
+) : KAbstractModule() {
+  override fun configure() {
+    install(FakeAwsEnvironmentModule())
+  }
+
+  @Provides @Singleton
+  fun providesAmazonDynamoDB(): AmazonDynamoDB {
+    val dynamoDb = AmazonDynamoDBClientBuilder
+        .standard()
+        .withCredentials(AWSStaticCredentialsProvider(BasicAWSCredentials("key", "secret")))
+        .withEndpointConfiguration(
+            AwsClientBuilder.EndpointConfiguration(
+                "http://localhost:8000", // TODO: inject from config
+                Regions.US_WEST_1.toString()
+            )
+        )
+        .build()
+
+    entities.forEach {
+      createTableForEntity(
+          dynamoDb,
+          it.second,
+          it.first
+      )
+    }
+
+    return dynamoDb
+  }
+
+  @Provides @Singleton
+  fun providesDynamoDBmapper(dynamoDb: AmazonDynamoDB): DynamoDBMapper {
+    return RealDynamoDbModule.dynamoDbMapper(dynamoDb)
+  }
+
+  companion object {
+    fun createTableForEntity(
+      amazonDynamoDB: AmazonDynamoDB,
+      entity: KClass<*>,
+      tableName: String
+    ) {
+      val tableRequest = DynamoDBMapper(amazonDynamoDB)
+          .generateCreateTableRequest(entity.java)
+          .withTableName(tableName)
```

Comment:
I believe DynamoDBMapper#describeTable gives you the table name

Reply:
I cant seem to see this in intelliSense. And how do you descrbe a table which doesn't exist?

Reply:
I was referring to [this method](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/dynamodbv2/datamodeling/DynamoDBTableMapper.html#describeTable--)

However I did not use it on my branch. I had the following working (with out specifying table names)
```kotlin
@Singleton
internal class CreateTablesService @Inject constructor(
  private val dynamoDb: AmazonDynamoDB,
  private val dynamoDbMapper: DynamoDBMapper,
  private val tables: Set<DynamoDbTable>
) : AbstractIdleService() {

  override fun startUp() {
    for (tableName in dynamoDb.listTables().tableNames) {
      dynamoDb.deleteTable(tableName)
    }

    val provisionedThroughput = ProvisionedThroughput(1000L, 1000L)
    for (table in tables) {
      try {
        val createTableRequest = dynamoDbMapper.generateCreateTableRequest(table.tableClass.java)
            .withProvisionedThroughput(provisionedThroughput)
        dynamoDb.createTable(createTableRequest)
      } catch (e: ResourceInUseException) {
        logger.error(e) { "Cannot create table." }
      }
    }
  }

  override fun shutDown() {
    dynamoDb.shutdown()
  }
}
/**
 * [DynamoDbTable] is a wrapper class that allows for unqualified binding of dynamo table classes
 * without collision.
 */
internal data class DynamoDbTable(val tableClass: KClass<*>)
```